opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -12,141 +12,123 @@
-        print(Context)
-
-    if __name__ == '__main__':
-        asyncio.run(main())
-"""
-
-from .base_context import BaseRuntimeContext
-
-__all__ = ["Context"]
-
-try:
-    from .async_context import AsyncRuntimeContext
-
-    Context = AsyncRuntimeContext()  # type: BaseRuntimeContext
-except ImportError:
-    from .thread_local_context import ThreadLocalRuntimeContext
-
-    Context = ThreadLocalRuntimeContext()
+import logging
+import typing
+from os import environ
+
+from pkg_resources import iter_entry_points
+
+from opentelemetry.context.context import Context
+
+logger = logging.getLogger(__name__)
+_CONTEXT = None  # type: typing.Optional[Context]
+
+
+def _copy_context(context: typing.Optional[Context]) -> Context:
+    if context is not None:
+        return context.copy()
+    return get_current().copy()
+
+
+def create_key(key: str) -> "object":
+    # FIXME Implement this
+    raise NotImplementedError
+
+
+def get_value(key: str, context: typing.Optional[Context] = None) -> "object":
+    """
+    To access the local state of an concern, the Context API
+    provides a function which takes a context and a key as input,
+    and returns a value.
+
+    Args:
+        key: The key of the value to retrieve.
+        context: The context from which to retrieve the value, if None, the current context is used.
+    """
+    if context is not None:
+        return context.get_value(key)
+    return get_current().get_value(key)
+
+
+def set_value(
+    key: str, value: "object", context: typing.Optional[Context] = None
+) -> Context:
+    """
+    To record the local state of a cross-cutting concern, the
+    Context API provides a function which takes a context, a
+    key, and a value as input, and returns an updated context
+    which contains the new value.
+
+    Args:
+        key: The key of the entry to set
+        value: The value of the entry to set
+        context: The context to copy, if None, the current context is used
+    """
+    new_context = _copy_context(context)
+    new_context.set_value(key, value)
+    return new_context
+
+
+def remove_value(
+    key: str, context: typing.Optional[Context] = None
+) -> Context:
+    """
+    To remove a value, this method returns a new context with the key cleared.
+    Note that the removed value still remains present in the old context.
+
+    Args:
+        key: The key of the entry to remove
+        context: The context to copy, if None, the current context is used
+    """
+    new_context = _copy_context(context)
+    new_context.remove_value(key)
+    return new_context
+
+
+def get_current() -> Context:
+    """
+    To access the context associated with program execution,
+    the Context API provides a function which takes no arguments
+    and returns a Context.
+    """
+    global _CONTEXT  # pylint: disable=global-statement
+    if _CONTEXT is None:
+        # FIXME use a better implementation of a configuration manager to avoid having
```

Comment:
Couldn't you pass the name to "iter_entry_points" already? https://setuptools.readthedocs.io/en/latest/pkg_resources.html#convenience-api

Then you could use `_CONTEXT = next(iter_entry_points("opentelemetry_context", environ.get("OPENTELEMETRY_CONTEXT", "default_context")).load()()` instead of constructing a map.

Reply:
Good point! Updated.

Reply:
Did you push the update yet?

Reply:
Yes the code is below

Reply:
Ah sorry, it seems I jumped to an old revision in the diff.

Reply:
The same happened to me yesterday 😄